isminet/models/validators.py

```
@@ -53,6 +53,24 @@ def validate_ip(v: Optional[str]) -> Optional[str]:
+        return None
+    validated = []
+    for addr in v:
+        try:
+            IPv4Address(addr)
+            validated.append(addr)
+        except ValueError:
+            raise PydanticCustomError(
+                "invalid_ip",
+                "Invalid IPv4 address: {addr}",
+                {"addr": addr},
+            )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)